pkg/controller/controller_broker_test.go

```diff
@@ -377,40 +374,41 @@ func TestReconcileBrokerWithAuthError(t *testing.T) {
 }
 
 func TestReconcileBrokerWithReconcileError(t *testing.T) {
-	fakeKubeClient, fakeCatalogClient, _, testController, _ := newTestController(t)
+	fakeKubeClient, fakeCatalogClient, fakeBrokerClient, testController, _ := newTestController(t, getTestCatalogConfig())
 
 	broker := getTestBroker()
-	broker.Spec.AuthInfo = &v1alpha1.BrokerAuthInfo{
-		BasicAuthSecret: &v1.ObjectReference{
-			Namespace: "does_not_exist",
-			Name:      "auth-name",
-		},
-	}
 
 	fakeCatalogClient.AddReactor("create", "serviceclasses", func(action clientgotesting.Action) (bool, runtime.Object, error) {
 		return true, nil, errors.New("error creating serviceclass")
 	})
 
 	testController.reconcileBroker(broker)
 
+	brokerActions := fakeBrokerClient.Actions()
+	assertNumberOfBrokerActions(t, brokerActions, 1)
+	assertGetCatalog(t, brokerActions[0])
+
 	actions := fakeCatalogClient.Actions()
-	assertNumberOfActions(t, actions, 1)
+	assertNumberOfActions(t, actions, 2)
 
-	updatedBroker := assertUpdateStatus(t, actions[0], broker)
+	createSCAction := actions[0].(clientgotesting.CreateAction)
+	createdSC, ok := createSCAction.GetObject().(*v1alpha1.ServiceClass)
+	if !ok {
+		t.Fatalf("couldn't convert to a ServiceClass: %+v", createSCAction.GetObject())
+	}
+	if e, a := getTestServiceClass(), createdSC; !reflect.DeepEqual(e, a) {
+		t.Fatalf("unexpected diff for created ServiceClass: %v,\n\nEXPECTED: %+v\n\nACTUAL:  %+v", diff.ObjectReflectDiff(e, a), e, a)
+	}
+	updatedBroker := assertUpdateStatus(t, actions[1], broker)
 	assertBrokerReadyFalse(t, updatedBroker)
 
 	kubeActions := fakeKubeClient.Actions()
-	assertNumberOfActions(t, kubeActions, 1)
-
-	getAction := kubeActions[0].(clientgotesting.GetAction)
-	if e, a := "get", getAction.GetVerb(); e != a {
-		t.Fatalf("Unexpected verb on action; expected %v, got %v", e, a)
-	}
+	assertNumberOfActions(t, kubeActions, 0)
 
 	events := getRecordedEvents(testController)
 	assertNumEvents(t, events, 1)
 
-	expectedEvent := api.EventTypeWarning + " " + errorAuthCredentialsReason + " " + "Error getting broker auth credentials for broker \"test-broker\": auth secret didn't contain username"
+	expectedEvent := api.EventTypeWarning + " " + errorSyncingCatalogReason + ` Error reconciling serviceClass "test-serviceclass" (broker "test-broker"): error creating serviceclass`
```

Comment:
sure - this is basically how every other test does it, and i want to clean up every other test too, follow-up okay?